cmd/soroban-rpc/internal/daemon/metrics.go

```diff
@@ -40,3 +47,56 @@ func (d *Daemon) MetricsRegistry() *prometheus.Registry {
 func (d *Daemon) MetricsNamespace() string {
 	return prometheusNamespace
 }
+
+type CoreClientWithMetrics struct {
+	stellarcore.Client
+	submitMetric  *prometheus.SummaryVec
+	opCountMetric *prometheus.SummaryVec
+}
+
+func newCoreClientWithMetrics(client stellarcore.Client, registry *prometheus.Registry) *CoreClientWithMetrics {
+	submitMetric := prometheus.NewSummaryVec(prometheus.SummaryOpts{
```

Comment:
Should we aim to match the metrics we emit from horizon txsub? wdyt?
https://github.com/stellar/go/blob/master/services/horizon/internal/txsub/system.go#L410

Reply:
No, the concern is that it seems like horizon tracks these a bit more in-depth. open/buffered/failed, etc. I like using the response status, but we don't currently get visibility into the actual txsub queueing. But I guess we don't have any queueing here, so txsub is a bit simpler... 🤔

Reply:
(And that it would be nice to match naming and metrics conventions for consistency)